test/runners.jl

```diff
@@ -123,6 +123,30 @@ end
         end
     end
 
+    # Test that we get no warnings when compiling without linking and when building a shared lib with clang
+    @testset "Clang - $(platform)" for platform in platforms
+        mktempdir() do dir
+            ur = preferred_runner()(dir; platform=platform)
+            iobuff = IOBuffer()
+            test_c = """
+            int test(void) {
+                return 0;
+            }
+            """
+            test_script = """
+            set -e
+            echo '$(test_c)' > test.c
+            clang -Werror -c test.c
+            clang -Werror -shared test.c -o test.\${dlext}
+            """
+            cmd = `/bin/bash -c "$(test_script)"`
+            @test run(ur, cmd, iobuff; tee_stream=devnull) broken=Sys.iswindows(platform)
```

Comment:
May also be good to explain that Windows is broken because of #248

Reply:
Yeah, it needs some polish still. I just want to pass tests first ;)

Reply:
Has armv7/6 ever worked on clang? Or do I just disable the test on that platform?

Reply:
It fails on musl only. Also windows passes this test apparently :)

Reply:
```console
% julia --compile=min -e 'using BinaryBuilderBase; BinaryBuilderBase.runshell(Platform("armv7l", "linux"; libc="musl"))'
sandbox:${WORKSPACE} # echo 'int test(){return 0;}' | clang -x c - -c -o test.o
clang-13: warning: argument unused during compilation: '-rtlib=libgcc' [-Wunused-command-line-argument]
clang-13: warning: argument unused during compilation: '-stdlib=libstdc++' [-Wunused-command-line-argument]
sandbox:${WORKSPACE} # echo 'int test(){return 0;}' | clang -x c - -shared -o test.so
/opt/arm-linux-musleabihf/bin/arm-linux-musleabihf-ld: cannot find crtbeginS.o: No such file or directory
/opt/arm-linux-musleabihf/bin/arm-linux-musleabihf-ld: cannot find -lgcc
/opt/arm-linux-musleabihf/bin/arm-linux-musleabihf-ld: cannot find -lgcc_s
/opt/arm-linux-musleabihf/bin/arm-linux-musleabihf-ld: cannot find -lgcc
/opt/arm-linux-musleabihf/bin/arm-linux-musleabihf-ld: cannot find -lgcc_s
/opt/arm-linux-musleabihf/bin/arm-linux-musleabihf-ld: cannot find crtendS.o: No such file or directory
clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
sandbox:${WORKSPACE} # qfind /opt/${target}/${target} -name 'libgcc_s.*'
/opt/arm-linux-musleabihf/arm-linux-musleabihf/lib/libgcc_s.so
/opt/arm-linux-musleabihf/arm-linux-musleabihf/lib/libgcc_s.so.1
```
uhm, the linker seems subtly broken, no idea why, libraries are of course there (otherwise gcc wouldn't work either).  Mind opening another issue and mark also those platforms as broken?